crates/install-wheel-rs/src/linker.rs

```diff
@@ -416,11 +416,12 @@ fn hardlink_wheel_files(
                         // Removing and recreating would lead to race conditions.
                         let tempdir = tempdir_in(&site_packages)?;
                         let tempfile = tempdir.path().join(entry.file_name());
-                        if fs::hard_link(path, &tempfile).is_err() {
+                        if fs::hard_link(path, &tempfile).is_ok() {
+                            fs_err::rename(&tempfile, &out_path)?;
+                        } else {
                             fs::copy(path, &out_path)?;
                             attempt = Attempt::UseCopyFallback;
                         }
-                        fs_err::rename(&tempfile, &out_path)?;
```

Comment:
Is this really changed behavior than it was before?
IMO you could handle `fs_err::rename` error instead:

```rs
                        if fs::hard_link(path, &tempfile).is_err() {
                            fs::copy(path, &out_path)?;
                            attempt = Attempt::UseCopyFallback;
                        }
                        if fs_err::rename(&tempfile, &out_path).is_err() {
                            fs::copy(path, &out_path)?;
                            attempt = Attempt::UseCopyFallback;
                        }
```

Reply:
Yeah, it's a change, right? Before, we did `fs_err::rename(&tempfile, &out_path)?;` unequivocally. Now, we only do it in the event that the hard link fails.

Reply:
> we did `fs_err::rename(&tempfile, &out_path)?;` unequivocally

Sorry, my bad.
I thought it will send return from if-block (at python we do mostly if-return instead of if-else 😄).